src/extension/features/budget/budget-spending-goal/index.js

```diff
@@ -0,0 +1,181 @@
+import { Feature } from 'toolkit/extension/features/feature';
+import { isCurrentRouteBudgetPage } from 'toolkit/extension/utils/ynab';
+import { getSpendingGoals, setSpendingGoals } from 'toolkit/extension/utils/spendinggoals';
+
+export class BudgetSpendingGoal extends Feature {
+  shouldInvoke() {
+    return isCurrentRouteBudgetPage();
+  }
+
+  injectCSS() {
+    return require('./index.css');
+  }
+
+  invoke() {
+    $('.budget-table-row')
+      .not('.budget-table-uncategorized-transactions')
+      .not('.is-debt-payment-category')
+      .not('.is-master-category')
+      .each((index, element) => {
+        const fundingAllocations = getSpendingGoals();
+        const subcategoryID = $(element).attr('data-entity-id');
+        const isGoalStored = fundingAllocations.contains(subcategoryID);
+        this.updateAvailableUI(element, isGoalStored);
+      });
+
+    const element = $('.budget-table-row.is-sub-category.is-checked').not(
+      '.is-debt-payment-category'
+    );
+    const subcategoryID = $(element).attr('data-entity-id');
+    if ($('.ynab-radiobutton').find("button.is-checked[data-value='MF']").length) {
+      // check if the Monthly funding goal is selected.
+      if (this.doesMaxLabelExist()) {
+        $('.goallabel').show();
+        if (getSpendingGoals().contains(subcategoryID)) {
+          $('input.spend-goal').prop('checked', true);
+        }
+      } else {
+        $('.ynab-radiobutton').append(
+          "<label class='goallabel'>" +
+            " <input class='spend-goal' type='checkbox' value='MG'>" +
+            ' Is max spending goal.' +
+            '</label> '
+        );
+        this.updateCategorySpendGoal();
+      }
+    } else if (this.doesMaxLabelExist()) {
+      $('.goallabel').hide();
+      this.updateCategorySpendGoal();
+    }
+
+    // When the goal inspector button is cliecked update the available UI.
+    $('button.button-primary').on(
+      'click',
+      this.updateAvailableUI(element, getSpendingGoals().contains(subcategoryID))
+    );
+
+    const goalstate = $(element).find(
+      '.ynab-new-budget-available-number.js-budget-available-number.user-data'
+    )[0];
+    if ($(goalstate).hasClass('positive')) {
+      this.udpateGoalChart('positive');
+      const goalElement = document.getElementsByClassName('goal-progress-container');
+      if ($(goalElement).hasClass('spending-goal')) {
+        $('.goal-progress-chart')
+          .find('.percent-label')
+          .text('Spent');
+      }
+    } else if ($(goalstate).hasClass('cautious')) {
+      // TODO: update goal chart to reflect overspending.
+      this.udpateGoalChart('cautious');
```

Comment:
should we be reflecting overspending in the available UI as well ?

Reply:
The available column gets updated previously to show overspending or positive. This section is a WIP. Here is where I would like to change the goal chart to show a percent of over spent rather than the current check mark. I have started working on this but dealing with all the variations of currencies is more challenging than I had thought. I think this would be a nice feature but I am not sure it is absolutely necessary. What you think?